CHANGELOG.md

```diff
@@ -1,6 +1,7 @@
 ## vNext (TBD)
 
 ### Enhancements
+* Added `SyncConfiguration.cancelWaitOnNonFatalError`. When set to true, all async operations (such as opening the realm with `await Realm.open`) will fail when a non-fatal error, such as a timeout, occurs.  
```

Comment:
Not sure if this is formulated properly 😁 
- I am unsure wether that's a correct terminology for a promise (it fails?)
- Also I don't know if there are other methods/promises that could fail because of this new parameter. In .NET we have also methods to asynchronously waiting for all downloads/uploads, but it seems that they don't exist in JS.

Reply:
`I am unsure wether that's a correct terminology for a promise (it fails?)`

I guess that would be `reject`? But personally I think it's intuitive as it is now, might need input form other people as well..

Reply:
> methods to asynchronously waiting for all downloads/uploads

We have `realm.syncSession.downloadAllServerChanges` and `realm.syncSession.uploadAllLocalChanges`.  Each take a timeout as a param

Reply:
https://www.mongodb.com/docs/realm-sdks/js/latest/Realm.App.Sync.Session.html